Add a form wrapper for better UX (#41304)
mdn/content: files/en-us/learn_web_development/core/scripting/dom_scripting/index.md

const input = document.querySelector("input");
const button = document.querySelector("button");

> button.addEventListener("click", (event) => {

Reviewer: would it make sense to listen to the form's `submit` event instead of the button's `click` event?

To complete the exercise, follow the steps below, and make sure that the list behaves as described above.

> 1. To start with, download a copy of our [shopping-list.html](https://github.com/mdn/learning-area/blob/main/javascript/apis/document-manipulation/shopping-list.html) starting file and make a copy of it somewhere. You'll see that it has some minimal CSS, a form with a label, input, and button, and an empty list and {{htmlelement("script")}} element. You'll be making all your additions inside the script.

Reviewer: ```suggestion
1. To begin, download a copy of our [shopping-list.html](https://github.com/mdn/learning-area/blob/main/javascript/apis/document-manipulation/shopping-list.html) starting file and make a copy of it somewhere. You'll see that it has some minimal CSS, a form with a label, input, and button, an empty list, and a {{htmlelement("script")}} element. You'll be making all your additions inside the script.
```

2. Create three variables that hold references to the list ({{htmlelement("ul")}}), {{htmlelement("input")}}, and {{htmlelement("button")}} elements.
3. Create a [function](/en-US/docs/Learn_web_development/Core/Scripting/Functions) that will run in response to the button being clicked.

> 4. Inside the function body, start off by calling [`preventDefault()`](/en-US/docs/Web/API/Event/preventDefault). When wrapped in the form element, the input will trigger the form to submit when the user presses the <kbd>Enter</kbd> key. This will keep the form from refreshing the page and will add a new item to the list.
> 5. Continue by storing the current [value](/en-US/docs/Web/API/HTMLInputElement/value) of the input element in a variable.

Reviewer: Nice addition about `preventDefault()`!
Suggesting a slightly different flow..
```suggestion
4. Inside the function body, start by calling [`preventDefault()`](/en-US/docs/Web/API/Event/preventDefault). Since the input is wrapped in a form element, pressing the <kbd>Enter</kbd> key will trigger the form to submit. The call to `preventDefault()` will prevent the form from refreshing the page so a new item can be added to the list instead.
5. Continue by storing the current [value](/en-US/docs/Web/API/HTMLInputElement/value) of the input in a variable.
```

> 6. Next, empty the input element by setting its value to an empty string — `""`.
> 7. Create three new elements — a list item ({{htmlelement('li')}}), {{htmlelement('span')}}, and {{htmlelement('button')}}, and store them in variables.
> 8. Append the span and the button as children of the list item.
> 9. Set the text content of the span to the input element value you saved earlier, and the text content of the button to 'Delete'.

Reviewer: ```suggestion
6. Next, clear the input element by setting its value to an empty string (`""`).
7. Create three new elements — a list item ({{htmlelement('li')}}), a {{htmlelement('span')}}, and a {{htmlelement('button')}} — and store them in variables.
8. Append the span and button as children of the list item.
9. Set the text content of the span to the input value you saved earlier, and set the text content of the button to `Delete`.
```

> 10. Append the list item as a child of the list.
> 11. Attach an event handler to the delete button so that, when clicked, it will delete the entire list item (`<li>...</li>`).
> 12. Finally, use the [`focus()`](/en-US/docs/Web/API/HTMLElement/focus) method to focus the input element, ready for entering the next shopping list item.

Reviewer: ```suggestion
10. Append the list item to the list.
11. Attach an event handler to the **Delete** button so that, when clicked, it removes the entire list item (`<li>...</li>`).
12. Finally, use the [`focus()`](/en-US/docs/Web/API/HTMLElement/focus) method to focus the input element, so it's ready for entering the next shopping list item.
```
